stonesoup/models/transition/nonlinear.py

```diff
@@ -100,7 +95,8 @@ def function(self, state, noise=False, **kwargs) -> StateVector:
         sv1 = state.state_vector
         turn_rate = sv1[4, :]
         # Avoid divide by zero in the function evaluation
-        turn_rate[turn_rate == 0.] = np.finfo(float).eps
+        if turn_rate[0] == 0:
+            turn_rate = Matrix([np.finfo(float).eps])
```

Comment:
I think, like before, this should check the entire array and set any zero elements to eps.

Reply:
OK, I've reverted to the original code, but now with a check to ensure the Matrix is `float` before trying to fix zero elements.